Sources/MongoSwift/MongoCollection+FindAndModify.swift

```diff
@@ -29,15 +32,18 @@ extension MongoCollection {
      *
      * - Returns: A `CollectionType`, representing either the original document or its replacement,
      *      depending on selected options, or `nil` if there was no match.
+     *
      * - Throws: 
+     *   - `MongoError.invalidArgument` if any of the provided options are invalid
      *   - `MongoError.commandError` if there are any errors executing the command.
      *   - An `EncodingError` if `replacement` cannot be encoded to a `Document`
      *   - A `DecodingError` if the replaced document cannot be decoded to a `CollectionType` value
      */
     @discardableResult
     public func findOneAndReplace(filter: Document, replacement: CollectionType,
-                                  options: FindOneAndDeleteOptions? = nil) throws -> CollectionType? {
-        throw MongoError.commandError(message: "Unimplemented command")
+                                  options: FindOneAndReplaceOptions? = nil) throws -> CollectionType? {
+        let update = try BsonEncoder().encode(replacement)
```

Comment:
Why does `update` need to be encoded? Is this just for the off chance that it's not a Document so that we can later access `Document.data`? Looking at `BsonEncoder.encode()`, passing it a Document appears to be a no-op.